enhancements/hypershift/msi-enablement-for-aro-hcp.md

```diff
@@ -0,0 +1,149 @@
+  - "@enxebre"
+api-approvers: # In case of new or modified APIs or API extensions (CRDs, aggregated apiservers, webhooks, finalizers). If there is no API change, use "None"
+  - "None"
+creation-date: 
+  - "2024-08-02"
+last-updated: 
+  - "2024-08-02"
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - "https://issues.redhat.com/browse/OCPSTRAT-979"
+see-also:
+  - "ARO HCP Authentication Strategy at HCP and Data Plane - https://docs.google.com/document/d/1Z7N2LAnRlgSgrFjjl2absOnkGFsI2TMcbwaW_CA1qek/edit#heading=h.bupciudrwmna"
+replaces:
+superseded-by:
+---
+
+# Add HyperShift Override Environment Variable for Azure Managed Service Identity
+
+## Summary
+
+This enhancement proposes introducing an environment variable in the image registry, ingress, cloud network config, 
+and storage operators. This variable would allow overriding the Azure authentication strategy used by these operators to 
+leverage Azure managed service identity (MSI), regardless of the underlying cloud configuration.
+
+## Motivation
+
+In Azure Red Hat OpenShift (ARO) Hosted Control Plane (HCP), operators running in the control plane need to 
+authenticate using Azure managed service identities to communicate with cloud services. In contrast, the same operators 
+running on the data plane/guest cluster side use workload identity authentication. 
+
+### User Stories
+
+* [Support MSI authentication in cluster-ingress-operator](https://issues.redhat.com/browse/NE-1504)
+* [Support MSI authentication in cloud-network-config-controller](https://issues.redhat.com/browse/SDN-4450)
+* [Support MSI authentication in cluster-storage-operator](https://issues.redhat.com/browse/STOR-1748)
+* [Support MSI authentication in image-registry](https://issues.redhat.com/browse/IR-460)
+
+### Goals
+
+* Agreement from ingress, image registry, network, and storage representatives on a standard approach to authenticate with MSI for ARO HCP
+
+### Non-Goals
+
+* Implementing MSI for image registry, ingress, cloud network config, and storage operators outside the override.
+
+## Proposal
+
+We propose setting an environment variable, AZURE_MSI_AUTHENTICATION, upon deployment of image registry, ingress, cloud 
+network config, and storage operators in the control plane of an ARO HCP cluster. This variable will be checked by each 
+operator; if set, it will override the default authentication mechanism, using a managed identity to authenticate with 
+Azure cloud services instead.
+
+For operators with operands that they manage in the control plane, the operator would be responsible for propagating the 
+environment variable to those operands (if the operands need cloud access).
+
+### Workflow Description
+
+* HostedCluster control plane operator will set AZURE_MSI_AUTHENTICATION on deployment of image registry, ingress, cloud network config, and storage operators
```

Comment:
NIT: The `cloud network config` is a second level operator managed by CNO. I think it would be better to set the env variable on CNO and let it pass it down to `cloud network config`

Reply:
I will make an update to clarify this.